fsspec/implementations/tests/test_http.py

```diff
@@ -97,6 +97,36 @@ def test_list(server):
     assert out == [server + "/index/realfile"]
 
 
+def test_list_invalid_args(server):
+    with pytest.raises(TypeError):
+        h = fsspec.filesystem("http", use_foobar=True)
+        h.glob(server + "/index/*")
+
+
+def test_list_cache(server):
+    h = fsspec.filesystem("http", use_listings_cache=True)
+    out = h.glob(server + "/index/*")
+    assert out == [server + "/index/realfile"]
+
+
+def test_list_cache_with_expiry_time(server):
+    h = fsspec.filesystem("http", use_listings_cache=True, listings_expiry_time=30)
```

Comment:
You could test these three parametrised cases to actually trigger the condition: have a short listing expiry time and sleep past it; list more than 5 paths; make a new instance with/without the instance cache (one should already have its dircache populated, the other not).

Reply:
Hi Martin,

thank you very much for your suggestions, I tried to implement some ideas from them as good as possible. However, maybe I am still missing some details. If you see the tests should be improved once more, I am humbly asking for your guidance and to elaborate a bit more about the details I might have missed.

> have a short listing expiry time and sleep past it

I just added b1f4cad55 in order to bring in your idea of verifying the actual contents of the directory cache beyond the expiration time.

> make a new instance with/without the instance cache (one should already have its dircache populated, the other not)

Do you mean something along the lines of ca0a61e6?

> list more than 5 paths

I am not sure I am following here. The test is currently using the some response mock as the other tests are also using. Is there some mock which would yield more than 5 paths? Would it actually make any difference?

With kind regards,
Andreas.